distribution/lib/Standard/Base/0.0.0-dev/src/Any.enso

```diff
@@ -108,22 +108,7 @@ type Any
                  a = 7 * 21
                  a == 147
     == : Any -> Boolean
-    == self that =
-        # If there is No_Such_Conversion, then `self` and `that` are probably
-        # host or polyglot values, so we just compare them with the default comparator.
-        eq_self = Comparable.from self
-        eq_that = Comparable.from that
-        similar_type = Meta.is_same_object eq_self eq_that
-        case similar_type of
-            False -> False
-            True ->
-                case Meta.is_same_object eq_self Default_Comparator of
-                    # Shortcut for objects with Default_Comparator, because of the performance.
-                    True -> Comparable.equals_builtin self that
-                    False ->
-                        case eq_self.compare self that of
-                            Ordering.Equal -> True
-                            _ -> False
+    == self that = Comparable.equals_builtin self that
```

Comment:
Let's jump to `EqualsNode` builtin as soon as possible. First of all let's try to handle simple cases of primitive values in the `EqualsNode`. If the values aren't that simple, jump to `EqualsAtomNode` to compare atoms - either structurally or by _custom comparator_. Or just to (still huge) `EqualsComplexNode` to compare everything else.

Reply:
I guess it should be easy now to add a simple `Any.==` builtin that just delegates to `Comparable.equals_builtin`, thus fixing https://github.com/enso-org/enso/issues/5610. It can be left for another PR, though.

Reply:
Done in https://github.com/enso-org/enso/pull/6280/commits/1c7ac56b901672d05f336133a41eda9cc16e1e6b.

Reply:
@JaroslavTulach Could you pls also revert this commit https://github.com/enso-org/enso/pull/4067/commits/2bf52564904d427f937af4f895641270d2d3e38d that introduced these imports, that are now unnecessary? That will fully test whether #5610 is really fixed.

Reply:
04bdbffda2 - removal of the imports seems to work fine.